tensorflow/stream_executor/dnn.h

```diff
@@ -667,8 +667,31 @@ class PoolingDescriptor {
   std::vector<int64> strides_;
 };
 
-typedef int64 AlgorithmType;
-constexpr AlgorithmType kDefaultAlgorithm = -1;
+class AlgorithmType {
+  public:
+  typedef int64 Index;
+  AlgorithmType()
+      : algo_(kDefaultAlgorithm), tensor_ops_enabled_(false) {}
+  AlgorithmType(Index a, bool use_tensor_ops)
+      : algo_(a), tensor_ops_enabled_(use_tensor_ops) {}
+  bool is_default() const { return algo_ == kDefaultAlgorithm; }
+  bool operator==(const AlgorithmType& other) const {
```

Comment:
Suggest reordering member functions so that operator== doesn't interrupt the getters.

Reply:
Done